doc/Makefile

```diff
@@ -19,25 +19,68 @@ default: container-build
 # should not be used with `make -f`.
 ##
 
+CONTAINER_VERSION=1.1.1
+CONTAINER_NAME=openassetio-doc-build
+
+##
+## Container settings
+##
+
+# The following variables are considered when using Docker
+# to build the documentation:
+#
+#  GITHUB_ORG: The organisation to pull containers from (and publish
+#              to). If unspecified, then `thefoundryvisionmongers` will
+#              be used. Note: org names must be lowercase.
+#
+# When publishing, the following additional vars MUST be set:
+#
+#  GITHUB_PAT:      A GitHub Personal Access Token with the following
+#                   permissions: write:packages.
+#  GITHUB_USERNAME: The GitHub username whose token is being used.
+#
+# eg:
+#  GITHUB_USER=<username> GITHUB_PAT=<token> make publish-docker
+#
+ifndef GITHUB_ORG
+GITHUB_ORG = thefoundryvisionmongers
+endif
+
+# Docker images are published to the GitHub container registry
+CONTAINER_PATH = ghcr.io/$(GITHUB_ORG)/$(CONTAINER_NAME)
+CONTAINER_TAG = $(CONTAINER_PATH):$(CONTAINER_VERSION)
 
 ##
 ## PHONY TARGETS
 ##
 
-.PHONY: clean clean-html clean-tooling clean-docker container-build deploy docker-image html todo tooling
+.PHONY: clean clean-html clean-tooling clean-docker container-build deploy docker-image html \
+        publish-docker tag-docker todo tooling
 
 #
 # Container build - Runs this Makefile via docker
 #
 
-container-build: docker-image
-	docker run --rm -v `pwd`/../:/src openassetio-doc-build bash -c 'make -C /src/doc html'
+container-build:
+	docker run --rm -v `pwd`/../:/src $(CONTAINER_TAG) bash -c 'make -C /src/doc html'
+
+# Targets for building/tagging/publishing the container
 
 docker-image: Dockerfile
-	docker build . -t openassetio-doc-build
+	docker build . -t $(CONTAINER_NAME)
```

Comment:
[consider] We could use `CONTAINER_TAG` throughout, then there is no need for a separate `tag-docker` step.

I hit this because when I tried to run `make` it failed because I didn't have the tagged image.  So I tried `make docker-image` (because I didn't read the updated docs :roll_eyes:) and of course it still didn't work because the image wasn't tagged.

I think the `CONTAINER_NAME` image is now superfluous, since it's never used except for tagging, which  "copies" the image (also see comment re. `clean-docker`).

Reply:
Thats a much better idea, thanks. [Done](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/156/commits/4cf8f40fa1c23d331ea63d10cb5e6dfed211988c).